### PR TITLE
Feature/alcatel

### PR DESCRIPTION
**Pull request by julsemaan**
# Description

Adds support and documentation for Alcatel switches
# Impacts

Even more switches support !
# Delete branch after merge

YES
# NEWS file entries
## New Features
- Added support for Mac Authentication and 802.1x to Alcatel switches

### CODE REVIEW

File: docs/PacketFence_Network_Devices_Configuration_Guide.asciidoc [hunk 1]
```diff
@@ -352,6 +352,192 @@ This switch does not support port-security.
...
+    uplink_dynamic=0
+    uplink=1001
+    RoleMap=Y
+    VlanMap=N
+    registrationRole=unreg
+    isolationRole=unreg
+    defaultRole=employee
+    guestRole=guest
+
+802.1x
```
Comment (whitx) [inline docs/PacketFence_Network_Devices_Configuration_Guide.asciidoc]: Missing:
vlan port mobile interface

File: docs/PacketFence_Network_Devices_Configuration_Guide.asciidoc [hunk 2]
```diff
@@ -352,6 +352,192 @@ This switch does not support port-security.
...
+++++++
+
+First, make sure you followed the steps above in 'Global configuration'
+
+You will need to configure the ports you want to do authentication on.
+
+    802.1x 1/2
```
Comment (whitx) [inline docs/PacketFence_Network_Devices_Configuration_Guide.asciidoc]: Should be: vlan port < interface > 802.1x enable

File: docs/PacketFence_Network_Devices_Configuration_Guide.asciidoc [hunk 3]
```diff
@@ -352,6 +352,192 @@ This switch does not support port-security.
...
+
+    vlan 2
+    vlan 5
+    vlan 20
+    vlan 100
+
+Next, configure the RADIUS server to be PacketFence
+
+    aaa radius-server "packetfence" host 192.168.1.5 key useStrongerSecret
+    aaa device-authentication mac packetfence
```
Comment (whitx) [inline docs/PacketFence_Network_Devices_Configuration_Guide.asciidoc]: on 6250 it is aaa authentication instead of aaa device-authentication

**Comment by whitx** [comment]
Tested with 802.1x, MAB, VoIP on 6250, 6860

File: lib/pf/Switch/Alcatel.pm [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+    try {
+        my $connection_info = {
+            nas_ip => $send_disconnect_to,
+            secret => $self->{'_radiusSecret'},
+            LocalAddr => $management_network->tag('vip'),
+        };
+
+        $logger->debug("[$self->{'_ip'}] Network device supports roles. Evaluating role to be returned.");
+        my $roleResolver = pf::roles::custom->instance();
+        my $role = $roleResolver->getRoleForNode($mac, $self);
```
Comment (fdurand) [inline lib/pf/Switch/Alcatel.pm]: why ?

**Comment by julsemaan** [comment]
Ready to merge ?

**Pull request merged**